sql/core/src/test/scala/org/apache/spark/sql/internal/ExecutorSideSQLConfSuite.scala

```diff
@@ -125,6 +125,41 @@ class ExecutorSideSQLConfSuite extends SparkFunSuite with SQLTestUtils {
     val e = intercept[SparkException](dummyQueryExecution1.toRdd.collect())
     assert(e.getCause.isInstanceOf[NoSuchElementException])
   }
+
+  test("SPARK-30556 propagate local properties to subquery execution thread") {
+    withSQLConf("spark.sql.subquery.maxThreadThreshold" -> "1") {
+      Seq(true)
```

Comment:
we should wrap the test with `withTempView`, which removes the temp view at the end.

Reply:
updated